### PR TITLE
updating k8s client version to 1.30 and go version to 1.22.2

### PR DESCRIPTION
**Pull request by jaidevmane**
*Issue #, if available:*

*Description of changes:*

Updating the k8s client libraries to 1.30, and updating go version to 1.22.1.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by DanielCKennedy — COMMENTED** [review]
k8s client libraries update to 1.30 looks good

**Comment by jaidevmane** [comment]
Picked up latest version of go-runner from `https://gallery.ecr.aws/eks-distro/kubernetes/go-runner`

**Review by DanielCKennedy — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-ARG golang_image=public.ecr.aws/eks-distro-build-tooling/golang:1.21
+ARG golang_image=public.ecr.aws/eks-distro-build-tooling/golang:1.22
```
Comment (xdu31) [inline Dockerfile:1]: This is not necessary actually, because the ARG can be overridden

Comment (jaidevmane) [reply Dockerfile:1]: Thanks! @xdu31. Should I change it back? I have changed references in all places to maintain consistency.
Would it be possible to get an approval?

Comment (kmala) [reply Dockerfile:1]: I think its better to change it here instead of overriding when possible.

Comment (jaidevmane) [reply Dockerfile:1]: Resolving this thread.

**Review by kmala — APPROVED** [review]

**Pull request merged**